plugins/module_utils/crypto/support.py

```diff
@@ -201,12 +205,21 @@ def load_certificate(path, content=None, backend='cryptography'):
     except (IOError, OSError) as exc:
         raise OpenSSLObjectError(exc)
     if backend == 'pyopenssl':
-        return crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
+        if identify_pem_format(cert_content) or der_support_enabled is False:
+            return crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
+        elif der_support_enabled:
+            raise OpenSSLObjectError('Certificate in DER format is not supported by the pyopenssl backend.')
     elif backend == 'cryptography':
-        try:
-            return x509.load_pem_x509_certificate(cert_content, cryptography_backend())
-        except ValueError as exc:
-            raise OpenSSLObjectError(exc)
+        if identify_pem_format(cert_content) or der_support_enabled is False:
```

Comment:
```suggestion
        if der_support_enabled is False or identify_pem_format(cert_content):
```

Reply:
The order was intentional to preserve the original behavior (raise PEM-format exception) when specifying a certificate in DER format indicating der_support_enabled to true. When we set der_support_enabled to True is means we can provide certificate in PEM or DER format - both are supported. If we don't care about preserving the original behavior, then we could apply your suggestion but we need to change `or` to `and` to support both.
I hope that clarified why I did it. What do you recommend?

Reply:
Right now even if the caller provides `der_support_enabled == False`, `identify_pem_format()` will be executed, even though its return value will not be used. Please change the order so that `identify_pem_format()` will not be executed when `der_support_enabled == False`.

Your behavior introduces additional overhead then `der_support_enabled == False`, while mine restores the original behavior (which is do not run `identify_pem_format()`).